.github/workflows/pr.yaml

```diff
@@ -196,3 +196,29 @@ jobs:
             echo "$(git diff --cached licenses)"
             exit 1
           fi
+
+  generate-metadata:
```

Comment:
I mainly added this so that the generated yaml could be reviewed. @breedx-splk wdyt do we want to have the metadata yaml attached to pr/ci builds? Or just add it to releases?

Reply:
Is the intent for the yaml to live in the repo alongside the code, or to be published, or something else? On release seems fine, especially if the target consumer is a docs generator of some future design. I don't think docs will necessarily branch/rev like normal repos.

Reply:
ok, I'll remove this after @theletterf has had a chance to review the generated yaml

Reply:
FWIW, .NET is releasing the YAML as a release artifact.